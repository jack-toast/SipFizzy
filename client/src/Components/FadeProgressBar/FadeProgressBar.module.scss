.ProgressContainer {
  opacity: 0;

  transition: ease 150ms opacity;
  &.show {
    opacity: 1;
  }
}
