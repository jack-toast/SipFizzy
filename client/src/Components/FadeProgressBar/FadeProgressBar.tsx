import React from 'react';
import { LinearProgress } from '@material-ui/core';
import clsx from 'clsx';
import styles from './FadeProgressBar.module.scss';

export type Props = {
  active: boolean;
};

const FadeProgressBar: React.FC<Props> = ({ active }: Props) => {
  return (
    <div
      className={clsx(styles.ProgressContainer, {
        [styles.show]: active,
      })}
      data-testid="progress-container"
    >
      <LinearProgress />
    </div>
  );
};

export default FadeProgressBar;
