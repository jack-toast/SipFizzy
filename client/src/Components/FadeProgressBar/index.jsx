import React from 'react';
import PropTypes from 'prop-types';
import { LinearProgress } from '@material-ui/core';
import clsx from 'clsx';
import styles from './styles.module.scss';

const FadeProgressBar = ({ active }) => {
  return (
    <div
      className={clsx(styles.ProgressContainer, {
        [styles.show]: active,
      })}
    >
      <LinearProgress />
    </div>
  );
};

FadeProgressBar.propTypes = {
  active: PropTypes.bool.isRequired,
};

export default FadeProgressBar;
