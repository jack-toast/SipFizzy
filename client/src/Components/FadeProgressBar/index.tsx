import React from 'react';
import { LinearProgress } from '@material-ui/core';
import clsx from 'clsx';
import styles from './styles.module.scss';

type Props = {
  active: boolean;
  style?: React.CSSProperties;
};
const FadeProgressBar: React.FC<Props> = ({ active, style }: Props) => {
  return (
    <div
      style={{ ...style }}
      className={clsx(styles.ProgressContainer, {
        [styles.show]: active,
      })}
    >
      <LinearProgress />
    </div>
  );
};

export default FadeProgressBar;
