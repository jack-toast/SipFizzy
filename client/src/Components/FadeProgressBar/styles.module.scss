.ProgressContainer {
  opacity: 0;

  transition: ease 50ms opacity;
  &.show {
    opacity: 1;
  }
}
