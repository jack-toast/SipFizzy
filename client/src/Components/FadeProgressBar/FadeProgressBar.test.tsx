import React from 'react';
import { render } from '@testing-library/react';

import FadeProgressBar, { Props } from './FadeProgressBar';

function renderFadeProgressBar(props: Partial<Props> = {}) {
  const defaultProps: Props = {
    active: false,
  };
  return render(<FadeProgressBar {...defaultProps} {...props} />);
}

describe('<FadeProgressBar />', () => {
  test('should not display progress bar', async () => {
    const { findByTestId, container } = renderFadeProgressBar();
    const fadeProgressBar = await findByTestId('progress-container');
    expect(container).toMatchSnapshot();
  });
});
