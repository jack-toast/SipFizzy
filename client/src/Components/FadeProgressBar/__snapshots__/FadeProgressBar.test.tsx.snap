// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FadeProgressBar /> should not display progress bar 1`] = `
<div>
  <div
    class="ProgressContainer"
    data-testid="progress-container"
  >
    <div
      class="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-indeterminate"
      role="progressbar"
    >
      <div
        class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Indeterminate"
      />
      <div
        class="MuiLinearProgress-bar MuiLinearProgress-bar2Indeterminate MuiLinearProgress-barColorPrimary"
      />
    </div>
  </div>
</div>
`;
